packages/block-library/src/gallery/gallery.js

```diff
@@ -6,106 +6,82 @@ import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { RichText } from '@wordpress/block-editor';
+import {
+	RichText,
+	__experimentalUseInnerBlocksProps as useInnerBlocksProps,
+} from '@wordpress/block-editor';
 import { VisuallyHidden } from '@wordpress/components';
-import { __, sprintf } from '@wordpress/i18n';
+import { __ } from '@wordpress/i18n';
 import { createBlock } from '@wordpress/blocks';
+import { useEffect, useRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import GalleryImage from './gallery-image';
 import { defaultColumnsNumber } from './shared';
 
+const allowedBlocks = [ 'core/image' ];
+
 export const Gallery = ( props ) => {
 	const {
 		attributes,
 		isSelected,
 		setAttributes,
-		selectedImage,
 		mediaPlaceholder,
-		onMoveBackward,
-		onMoveForward,
-		onRemoveImage,
-		onSelectImage,
-		onDeselectImage,
-		onSetImageAttributes,
-		onFocusGalleryCaption,
 		insertBlocksAfter,
 		blockProps,
 	} = props;
 
 	const {
+		imageCount,
 		align,
-		columns = defaultColumnsNumber( attributes ),
+		columns = defaultColumnsNumber( imageCount ),
 		caption,
 		imageCrop,
-		images,
 	} = attributes;
+	const galleryRef = useRef();
+	const { children, ...innerBlocksProps } = useInnerBlocksProps( blockProps, {
+		allowedBlocks,
+		orientation: 'horizontal',
+		renderAppender: false,
+		__experimentalLayout: { type: 'default', alignments: [] },
+	} );
+
+	useEffect( () => {
+		if ( galleryRef.current && isSelected ) {
+			galleryRef.current.parentElement.focus();
```

Comment:
It was initially there to resolve this issue - https://github.com/WordPress/gutenberg/pull/25940#issuecomment-707832454, but on checking it isn't actually doing anything now, I suspect because a lot has changed with using blockProps and innerBlockProps since adding that, so I have removed it now and will need to look for another solution to that problem